src/cma.ts

```diff
@@ -281,15 +238,18 @@ async function runCumulusTask(taskFunction, cumulusMessage,
     cmaStdin.write('\n<EOC>\n');
     const createNextEventOutput = await getCmaOutput(rl, errorObj);
     cmaStdin.write('\n<EXIT>\n');
+    if (isLoadNestedEventInput(createNextEventOutput)) {
+      throw new Error(`Invalid typing recieved from
+      createNextEventOutput: ${JSON.stringify(createNextEventOutput)}`);
+    }
     return createNextEventOutput;
-  }
-  catch (error) {
-    if (error.name && error.name.includes('WorkflowError')) {
+  } catch (error) {
+    if (error.name && error.name.includes('WorkflowError') && (!isCMAMessage(cumulusMessage))) {
+      const returnObject = cumulusMessage;
```

Comment:
What is this doing?

Reply:
I think this is some legacy behavior where errors named `WorkflowError` don't fail the workflow

Reply:
Unless I'm missing something, though, `returnObject` is created and modified, but never used?

Reply:
So this *is* bad, but not in the way you think.   It's not deepcopying cumulusMessage so it works due to mutation, I missed the update to the return object.  

3ad3447

However ... that doesn't really make things better as runCumulusTask can still mutate cumulusMessage in case of error. 

So:

f005a17

Reply:
.... and I misread that.    Cloning isn't needed: 547e2cf